test/test_mixins/test_hash_commands.py

```diff
@@ -312,3 +318,55 @@ def test_hrandfield(r: redis.Redis):
+        (datetime.timedelta(seconds=100), BASE_TIME + 50, False, False, False, False, HexpireResult.SUCCESS),
+        # NX
+        (BASE_TIME + 100, None, True, False, False, False, HexpireResult.SUCCESS),
+        (datetime.timedelta(seconds=100), None, True, False, False, False, HexpireResult.SUCCESS),
+        (BASE_TIME + 100, BASE_TIME + 50, True, False, False, False, HexpireResult.CONDITION_UNMET),
+        (datetime.timedelta(seconds=100), BASE_TIME + 50, True, False, False, False, HexpireResult.CONDITION_UNMET),
+        # XX
+        (BASE_TIME + 100, None, False, True, False, False, HexpireResult.CONDITION_UNMET),
+        (datetime.timedelta(seconds=100), None, False, True, False, False, HexpireResult.CONDITION_UNMET),
+        (BASE_TIME + 100, BASE_TIME + 50, False, True, False, False, HexpireResult.SUCCESS),
+        (datetime.timedelta(seconds=100), BASE_TIME + 50, False, True, False, False, HexpireResult.SUCCESS),
+        # GT
+        (BASE_TIME + 100, None, False, False, True, False, HexpireResult.CONDITION_UNMET),
+        (datetime.timedelta(seconds=100), None, False, False, True, False, HexpireResult.CONDITION_UNMET),
+        (BASE_TIME + 100, BASE_TIME + 50, False, False, True, False, HexpireResult.SUCCESS),
+        (datetime.timedelta(seconds=100), BASE_TIME + 50, False, False, True, False, HexpireResult.SUCCESS),
+        (BASE_TIME + 100, BASE_TIME + 100, False, False, True, False, HexpireResult.CONDITION_UNMET),
+        (datetime.timedelta(seconds=100), BASE_TIME + 100, False, False, True, False, HexpireResult.CONDITION_UNMET),
+        (BASE_TIME + 100, BASE_TIME + 200, False, False, True, False, HexpireResult.CONDITION_UNMET),
+        (datetime.timedelta(seconds=100), BASE_TIME + 200, False, False, True, False, HexpireResult.CONDITION_UNMET),
+        # LT
+        (BASE_TIME + 100, None, False, False, False, True, HexpireResult.CONDITION_UNMET),
+        (datetime.timedelta(seconds=100), None, False, False, False, True, HexpireResult.CONDITION_UNMET),
+        (BASE_TIME + 100, BASE_TIME + 50, False, False, False, True, HexpireResult.CONDITION_UNMET),
+        (datetime.timedelta(seconds=100), BASE_TIME + 50, False, False, False, True, HexpireResult.CONDITION_UNMET),
+        (BASE_TIME + 100, BASE_TIME + 100, False, False, False, True, HexpireResult.CONDITION_UNMET),
+        (datetime.timedelta(seconds=100), BASE_TIME + 100, False, False, False, True, HexpireResult.CONDITION_UNMET),
+        (BASE_TIME + 100, BASE_TIME + 200, False, False, False, True, HexpireResult.SUCCESS),
+        (datetime.timedelta(seconds=100), BASE_TIME + 200, False, False, False, True, HexpireResult.SUCCESS),
+    ]
+)
+@patch.object(time, "time", BASE_TIME)
+def test_hexpire(r: redis.Redis, current_time: float, expiration: Union[int, datetime.timedelta], preset_expiration: Union[float, None], nx: bool, xx: bool, gt: bool, lt: bool, expected_result: int) -> None:
```

Comment:
since these commands do not exist before redis 7.4.0, you need to add `@pytest.mark.min_server("7.4")` to the test, not sure whether before/after the `@pytest.mark.parametrize` makes a difference.